cli/src/pixl_cli/_database.py

```diff
@@ -38,85 +37,91 @@
 engine = create_engine(url)
 
 
-def filter_exported_or_add_to_db(messages: list[Message], project_slug: str) -> list[Message]:
+def filter_exported_or_add_to_db(messages_df: pd.DataFrame, project_slug: str) -> pd.DataFrame:
     """
-    Filter exported images for this project, and adds missing extract or images to database.
+    Filter exported images for this project, and adds missing extract and images to database.
 
     :param messages: Initial messages to filter if they already exist
     :param project_slug: project slug to query on
     :return messages that have not been exported
     """
     PixlSession = sessionmaker(engine)
     with PixlSession() as pixl_session, pixl_session.begin():
-        extract, extract_created = _get_or_create_project(project_slug, pixl_session)
+        extract = pixl_session.query(Extract).filter(Extract.slug == project_slug).one_or_none()
+        if extract:
+            db_images_df = all_images_for_project(project_slug)
+            missing_images_df = _filter_existing_images(messages_df, db_images_df)
+            messages_df = _filter_exported_messages(messages_df, db_images_df)
+        else:
+            pixl_session.add(Extract(slug=project_slug))
+            extract = pixl_session.query(Extract).filter(Extract.slug == project_slug).one_or_none()
```

Comment:
do we need to query the session for it, or can we just create the extract and add it to the session, using it later down the line?

Reply:
ah yeah, this is also related to using `session.bulk_save_objects` - if you do `extract = Extract(slug=project_slug)` and then add this to the session, `extract.extract_id` doesn't exit. So you need to create it, add it to the session, and the query it to make sure you can access `extract.extract_id`

Reply:
nice ookay, might be worth adding a comment in the code when we set it just to be explicit that its required